### DISCUSSION TITLE
How to inquire the netmod being used in a MPICH installation

### DISCUSSION THREAD

**Discussion by vicentebolea** [pmodels/mpich#6331] [Q&A] [answered]
How to inquire the netmod being used in a MPICH installation when developing a C/C++/Fortran Application, is there any DEFINE or internal function that hints you the devices/netmod with which MPICH was built.

**Comment by raffenet** [comment, marked as answer]
Try `MPI_Get_library_version`. The lines starting with `MPICH Device:` and `MPICH configure:` should contain the device and library configuration.

**Reply by vicentebolea** [reply]
That worked well, thanks!

**Reply by raffenet** [reply]
Just FYI, MPICH also installs a binary file `mpichversion` which displays the same information and has various filtering options.
